ibm_pi_host_group: Update crashes with "interface {} is *schema.Set, not []interface {}"

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform CLI and Terraform IBM Provider Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

### Affected Resource(s)


* ibm_pi_host_group (PowerVS)

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

terraform-provider-ibm: v 1.88.2
I also attempted 1.88.1, 1.88.0, and 1.75.2 (which was before update to the source file), all resulted in the same error.

```hcl
# Minimal repro (redacted)
resource "ibm_pi_host_group" "example" {
  pi_cloud_instance_id = var.pi_cloud_instance_id

  # Host group name
  pi_name = "test-host-group"

  # (A) Hosts in the group
  pi_hosts = [
    {
      display_name = "host-1"
      sys_type     = "s922"
    }
  ]

  # (B) Share with a secondary workspace – toggling this triggers Update
  pi_secondaries = [
    {
      name      = "test-host-group-secondary"
      workspace = var.secondary_workspace_id
    }
  ]
}
```
### Debug Output

https://gist.github.com/jreeves-lus/b9f3330a4d3e5bea4a63001ddf86fde0

### Panic Output

https://gist.github.com/jreeves-lus/fe6f648d09031c4f3ad8cfb76b52eded

### Expected Behavior

Provider should handle the update of nested list attributes (pi_hosts, pi_secondaries) without crashing.

### Actual Behavior

Terraform fails with a provider panic during update.

### Steps to Reproduce

Steps to reproduce:
1. Apply the configuration above to create a host group.
2. Change one of the nested list values - 3.g. remove/add an entry in pi_secondaries or pi_hosts
3. Run terraform apply
4. Provider crashes during Update

```
module.powervs_workspace.ibm_pi_host_group.dedicated_hosts: Modifying... [id=<redacted>]
│ Error: Request cancelled
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.

Stack trace from the terraform-provider-ibm_v1.88.2 plugin:

panic: interface conversion: interface {} is *schema.Set, not []interface {}

goroutine 46 [running]:
github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/power.resourceIBMPIHostGroupUpdate({0x6e1f108, 0xc0004a17a0}, 0xc0043483f0, {0x628c940, 0xc000307c08})
        github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/power/resource_ibm_pi_host_group.go:237 +0xab1
...
Error: The terraform-provider-ibm_v1.88.2 plugin crashed!
```

### Important Factoids

Panic suggests an interface type assertion mismatch in the update handler (treats a *schema.Set as  []interface{}.  The resource and its implementation are in the PowerVS package (ResourceIBMPIHostGroup. 

### References

Documentation shows that pi_hosts and pi_secondaries are a List type:
https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/pi_host_group

Location of Update handler:
https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/ibm/service/power/resource_ibm_pi_host_group.go

* #0000

--- a/ibm/acctest/acctest.go
+++ b/ibm/acctest/acctest.go
@@ -270,6 +270,8 @@ var (
 	Pi_route_id                       string
 	Pi_sap_image                      string
 	Pi_sap_profile_id                 string
+	Pi_secondary_workspace_id_1       string
+	Pi_secondary_workspace_id_2       string
 	Pi_shared_processor_pool_id       string
 	Pi_snapshot_id                    string
 	Pi_spp_placement_group_id         string
@@ -1518,6 +1520,16 @@ func init() {
 	if Pi_host_group_id == "" {
 		Pi_host_group_id = ""
 		fmt.Println("[WARN] Set the environment variable PI_HOST_GROUP_ID for testing ibm_pi_host resource else it is set to default value ''")
+	}
+	Pi_secondary_workspace_id_1 = os.Getenv("PI_SECONDARY_WORKSPACE_ID_1")
+	if Pi_secondary_workspace_id_1 == "" {
+		Pi_secondary_workspace_id_1 = ""
+		fmt.Println("[WARN] Set the environment variable PI_SECONDARY_WORKSPACE_ID_1 for testing ibm_pi_host_group update else it is set to default value ''")
+	}
+	Pi_secondary_workspace_id_2 = os.Getenv("PI_SECONDARY_WORKSPACE_ID_2")
+	if Pi_secondary_workspace_id_2 == "" {
+		Pi_secondary_workspace_id_2 = ""
+		fmt.Println("[WARN] Set the environment variable PI_SECONDARY_WORKSPACE_ID_2 for testing ibm_pi_host_group update else it is set to default value ''")
 	}
 	Pi_host_id = os.Getenv("PI_HOST_ID")
 	if Pi_host_id == "" {

--- a/ibm/service/power/resource_ibm_pi_host_group.go
+++ b/ibm/service/power/resource_ibm_pi_host_group.go
@@ -51,12 +51,14 @@ func ResourceIBMPIHostGroup() *schema.Resource {
 					Schema: map[string]*schema.Schema{
 						Attr_DisplayName: {
 							Description:  "Name of the host chosen by the user.",
+							ForceNew:     true,
 							Required:     true,
 							Type:         schema.TypeString,
 							ValidateFunc: validation.NoZeroValues,
 						},
 						Attr_SysType: {
 							Description:  "System type.",
+							ForceNew:     true,
 							Required:     true,
 							Type:         schema.TypeString,
 							ValidateFunc: validation.NoZeroValues,
@@ -64,6 +66,7 @@ func ResourceIBMPIHostGroup() *schema.Resource {
 						Attr_UserTags: {
 							Description: "List of user tags attached to the resource.",
 							Elem:        &schema.Schema{Type: schema.TypeString},
+							ForceNew:    true,
 							Optional:    true,
 							Set:         schema.HashString,
 							Type:        schema.TypeSet,

--- a/ibm/service/power/resource_ibm_pi_host_group_test.go
+++ b/ibm/service/power/resource_ibm_pi_host_group_test.go
@@ -44,13 +44,95 @@ func testAccCheckIBMPIHostGroupConfigBasic(name, displayName string) string {
 			pi_cloud_instance_id = "%[1]s"
 			pi_hosts {
 				display_name = "%[2]s"
-				sys_type = "s1022"
+				sys_type = "s922"
 			}
 			pi_name = "%[3]s"
 		}
 	`, acc.Pi_cloud_instance_id, displayName, name)
 }
+func TestAccIBMPIHostGroupUpdateSecondaries(t *testing.T) {
+	name := fmt.Sprintf("tf_name_%d", acctest.RandIntRange(10, 100))
+	displayName := fmt.Sprintf("tf_display_name_%d", acctest.RandIntRange(10, 100))
+	hostGroupRes := "ibm_pi_host_group.hostGroup"
 
+	resource.Test(t, resource.TestCase{
+		Providers:    acc.TestAccProviders,
+		CheckDestroy: testAccCheckIBMPIHostGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				// Create host group with one secondary.
+				Config: testAccCheckIBMPIHostGroupConfigOneSecondary(name, displayName, acc.Pi_secondary_workspace_id_1),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckIBMPIHostGroupExists(hostGroupRes),
+					resource.TestCheckResourceAttr(hostGroupRes, "pi_name", name),
+					resource.TestCheckResourceAttr(hostGroupRes, "pi_secondaries.#", "1"),
+				),
+			},
+			{
+				// Add a second secondary workspace.
+				Config: testAccCheckIBMPIHostGroupConfigTwoSecondaries(name, displayName, acc.Pi_secondary_workspace_id_1, acc.Pi_secondary_workspace_id_2),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckIBMPIHostGroupExists(hostGroupRes),
+					resource.TestCheckResourceAttr(hostGroupRes, "secondaries.#", "2"),
+				),
+			},
+			{
+				// Remove the first secondary, keeping only the second.
+				Config: testAccCheckIBMPIHostGroupConfigOneSecondary(name, displayName, acc.Pi_secondary_workspace_id_2),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckIBMPIHostGroupExists(hostGroupRes),
+					resource.TestCheckResourceAttr(hostGroupRes, "secondaries.#", "1"),
+				),
+			},
+			{
+				// Remove all secondaries.
+				Config: testAccCheckIBMPIHostGroupConfigBasic(name, displayName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckIBMPIHostGroupExists(hostGroupRes),
+					resource.TestCheckResourceAttr(hostGroupRes, "secondaries.#", "0"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIBMPIHostGroupConfigOneSecondary(name, displayName, secondaryWorkspaceID string) string {
+	return fmt.Sprintf(`
+		resource "ibm_pi_host_group" "hostGroup" {
+			pi_cloud_instance_id = "%[1]s"
+			pi_hosts {
+				display_name = "%[2]s"
+				sys_type = "s922"
+			}
+			pi_name = "%[3]s"
+			pi_secondaries {
+				name      = "%[3]s-secondary"
+				workspace = "%[4]s"
+			}
+		}
+	`, acc.Pi_cloud_instance_id, displayName, name, secondaryWorkspaceID)
+}
+
+func testAccCheckIBMPIHostGroupConfigTwoSecondaries(name, displayName, secondaryWorkspaceID1, secondaryWorkspaceID2 string) string {
+	return fmt.Sprintf(`
+		resource "ibm_pi_host_group" "hostGroup" {
+			pi_cloud_instance_id = "%[1]s"
+			pi_hosts {
+				display_name = "%[2]s"
+				sys_type = "s922"
+			}
+			pi_name = "%[3]s"
+			pi_secondaries {
+				name      = "%[3]s-secondary-1"
+				workspace = "%[4]s"
+			}
+			pi_secondaries {
+				name      = "%[3]s-secondary-2"
+				workspace = "%[5]s"
+			}
+		}
+	`, acc.Pi_cloud_instance_id, displayName, name, secondaryWorkspaceID1, secondaryWorkspaceID2)
+}
 func testAccCheckIBMPIHostGroupExists(n string) resource.TestCheckFunc {
 
 	return func(s *terraform.State) error {

